platform/security/handler/security-handler-oidc/src/main/java/org/codice/ddf/security/oidc/client/OidcHandler.java

```diff
@@ -184,11 +185,22 @@ public HandlerResult getNormalizedToken(
         } else {
           LOGGER.error("OIDC HANDLER NOT CONFIGURED.");
           handlerResult.setStatus(HandlerResult.Status.NO_ACTION);
+          return handlerResult;
         }
       } catch (TechnicalException e) {
         // ignore
       }
 
+      // Authorization code flow
+      if (handlerConfiguration.getOidcConfiguration().getResponseType().equals("code")
+          && httpRequest.getParameter("code") != null) {
+
+        OidcAuthenticator authenticator =
+            new OidcAuthenticator(
+                handlerConfiguration.getOidcConfiguration(), handlerConfiguration.getOidcClient());
+        authenticator.validate(credentials, j2EContext);
```

Comment:
❓ Since this authenticator redirects the client, should we return a `HandlerResult` with the status of `REDIRECTED`?

Reply:
It doesn't redirect the user. The `OidcAuthenticator` makes a backend call and gets all the tokens.